src/components/media-views.js

```diff
@@ -837,3 +846,150 @@ AFRAME.registerComponent("media-image", {
     this.el.emit("image-loaded", { src: this.data.src, projection: projection });
   }
 });
+
+AFRAME.registerComponent("media-pdf", {
+  schema: {
+    src: { type: "string" },
+    projection: { type: "string", default: "flat" },
+    contentType: { type: "string" },
+    index: { default: 0 },
+    batch: { default: false }
+  },
+
+  init() {},
+
+  remove() {
+    if (this.data.batch && this.mesh) {
+      this.el.sceneEl.systems["hubs-systems"].batchManagerSystem.removeObject(this.mesh);
+    }
+
+    if (this.currentPageTextureIsRetained) {
+      const cacheKey = this.currentTextureCacheKey();
+      textureCache.release(cacheKey);
+      this.currentPageTextureIsRetained = false;
+    }
+  },
+
+  currentTextureCacheKey() {
+    // We ensure a unique texture for each PDF entity, because they are drawn over during pagination.
+    return `${this.el.object3D.uuid}_${this.canvas.width}_${this.canvas.height}`;
+  },
+
+  async update(oldData) {
+    let texture;
+    let ratio = 1;
+
+    try {
+      const { src, index } = this.data;
+      if (!src) return;
+
+      this.el.emit("pdf-loading");
+
+      if (!oldData.src || src !== oldData.src) {
+        const loadingSrc = this.data.src;
+        this.pdf = await pdfjs.getDocument(src);
+        if (loadingSrc !== this.data.src) return;
```

Comment:
This check should technically happen before changing `this.pdf`.. Though I am kind of thinking now that pdfs no long mutate `media-images` we just make all the `media-*` components (including media-loader) not support changing src after they have been initialized. Seems like it would drastically simplify the code.

Reply:
oh, good catch